tests/nose_runner.py

```diff
@@ -93,7 +109,339 @@ def generate_tasks_input():
+    img{{
+      width:100%;
+      max-width:800px;
+    }}
+    </style>
+    <h1>{header}</h1>
+    </head><body>
+    {body}
+    </body></html>
+    """
+
+    row_template = """
+    <tr>
+    <td align="center">Actual</td>
+    <td align="center">Expected</td>
+    <td align="center">Difference</td>
+    </tr>
+    <tr>
+    <td><img src="data:image/png;base64,{0}"></td>
+    <td><img src="data:image/png;base64,{1}"></td>
+    <td><img src="data:image/png;base64,{2}"></td>
+    </tr>
+    <tr><td align="center" colspan="3"><b>{3}</b><hr></td></tr>
+    """
+
+    table_template = """<table>{rows}</table>"""
+    rows = []
+
+    for test_name, images in failed_answers:
+        encoded_images = {}
+        for key in images:
+            with open(images[key], "rb") as img:
+                img_data = base64.b64encode(img.read()).decode()
+                encoded_images[key] = img_data
+
+        formatted_row = row_template.format(encoded_images["Actual"],
+                                            encoded_images["Expected"],
+                                            encoded_images["Difference"],
+                                            test_name)
+        rows.append(formatted_row)
+
+    body = table_template.format(rows='\n'.join(rows))
+    html = html_template.format(header="Failed Answer Tests", body=body)
+    return html
+
+def upload_to_curldrop(data, filename):
+    """Uploads file to yt's curldrop server
+
+    Uploads bytes `data` by the name `filename` to yt curldrop server.
+
+    Parameters
+    ----------
+    data : bytes
+        Content to be uploaded
+
+    filename : string
+        Name of file at curldrop's upload server
+
+    Returns
+    -------
+    requests.models.Response
+        Response returned by curldrop server
+
+    """
+    base_url = ytcfg.get("yt", "curldrop_upload_url")
+    upload_url = base_url + "/" + os.path.basename(filename)
+    response = requests.put(upload_url, data=data)
+    return response
+
+def upload_failed_answers(failed_answers):
+    """Uploads the result of failed answer tests
+
+    Uploads a html page of the failed answer tests.
+
+    Parameters
+    ----------
+    failed_answers : list of tuples (string, dict)
+        Collection of tuples where the first part is a string denoting the
+        test name, the second part is a dictionary that stores the actual,
+        expected and difference plot file locations of the test.
+
+    Returns
+    -------
+    requests.models.Response
+        Response as returned by the upload service
+
+    """
+    html = generate_webpage(failed_answers)
+    # convert html str to bytes
+    html = html.encode()
+    timestamp = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    filename = "failed_answers_{}.html".format(timestamp)
+
+    response = upload_to_curldrop(data=html, filename=filename)
+    return response
+
+def generate_missing_answers(answer_dir, missing_answers):
+    """Generate golden-answers
```

Comment:
Super tiny nitpick, but hyphenating doesn't make sense here or in the rest of the description.